consumers/writers/api/metrics.go

```diff
@@ -12,28 +12,61 @@ import (
 	"github.com/mainflux/mainflux/consumers"
 )
 
-var _ consumers.Consumer = (*metricsMiddleware)(nil)
+var _ consumers.AsyncConsumer = (*asyncMetricsMiddleware)(nil)
+var _ consumers.SyncConsumer = (*syncMetricsMiddleware)(nil)
 
-type metricsMiddleware struct {
-	counter  metrics.Counter
-	latency  metrics.Histogram
-	consumer consumers.Consumer
+type asyncMetricsMiddleware struct {
+	counter       metrics.Counter
+	latency       metrics.Histogram
+	asyncConsumer consumers.AsyncConsumer
 }
 
-// MetricsMiddleware returns new message repository
+// AsyncMetricsMiddleware returns new message repository
 // with Save method wrapped to expose metrics.
-func MetricsMiddleware(consumer consumers.Consumer, counter metrics.Counter, latency metrics.Histogram) consumers.Consumer {
-	return &metricsMiddleware{
-		counter:  counter,
-		latency:  latency,
-		consumer: consumer,
+func AsyncMetricsMiddleware(asyncConsumer consumers.AsyncConsumer, counter metrics.Counter, latency metrics.Histogram) consumers.AsyncConsumer {
```

Comment:
These async metrics make no sense since it's only one call and latency is very difficult to measure. Remove it.